osc/osc.go

```diff
@@ -573,6 +560,20 @@ func (s *Server) ReceivePacket(c net.PacketConn) (Packet, error) {
 	return s.readFromConnection(c)
 }
 
+type glue struct {
+	net.PacketConn
+}
+
+func (g glue) Read(buf []byte) (n int, err error) {
+	n, _, err = g.ReadFrom(buf)
+	if err == nil {
+		err = io.EOF
+	}
+	return
+}
+
+var g = &glue{}
```

Comment:
I'm not a fan of the whole glue thing, but it helps remove one race-condition (albeit, kinda replacing it with another).

The reason why I can't just use `net.Conn.Read()` is because it doesn't return `io.EOF` which `bytes.Buffer.ReadFrom()` requires in order to stop blocking.